UI/window-basic-properties.cpp

```diff
@@ -81,20 +81,10 @@ OBSBasicProperties::OBSBasicProperties(QWidget *parent, OBSSource source_)
 	OBSDataAutoRelease nd_settings = obs_source_get_settings(source);
 	obs_data_apply(oldSettings, nd_settings);
 
-	auto handle_memory = [](void *vp, obs_data_t *old_settings,
-				obs_data_t *new_settings) {
-		obs_source_t *source = reinterpret_cast<obs_source_t *>(vp);
-
-		obs_source_update(source, new_settings);
-
-		UNUSED_PARAMETER(old_settings);
-		UNUSED_PARAMETER(vp);
-	};
-
 	view = new OBSPropertiesView(
 		nd_settings.Get(), source,
 		(PropertiesReloadCallback)obs_source_properties,
-		(PropertiesUpdateCallback)handle_memory,
+		(PropertiesUpdateCallback) nullptr, // No special handling required for undo/redo
```

Comment:
With this change, the 'OK' button now crashes in both the Media Source and Browser Source because there's a `nullptr` check missing on `callback` in `UpdateSettings()` function that calls the callback.

https://github.com/obsproject/obs-studio/blob/45f737fb302810a225b17c9ab7fbec94147ae51e/UI/properties-view.hpp#L191

Unfortunately, adding a `nullptr` check exposes an issue where properties update, but the `Update` function for that source no longer gets called, so the Properties view displays the updated properties but the source doesn't apply the new properties (eg. new URL in Browser is not loaded, even after multiple Properties changes).

Reply:
I'm fixing it at the moment

Reply:
Should be fixed with https://github.com/obsproject/obs-studio/commit/89881147c7f31dd1869f3f877a4b1d342de863cf